src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -205,21 +206,20 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
             }
         }
 
-        private static FileStream CreateSharedBackingObjectUsingFile(Interop.Sys.MemoryMappedProtections protections, long capacity)
+        private static FileStream CreateSharedBackingObjectUsingFile(Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
             // We create a temporary backing file in TMPDIR.  We don't bother putting it into subdirectories as the file exists
             // extremely briefly: it's opened/created and then immediately unlinked.
             string path = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
 
-            FileAccess access =
-                (protections & (Interop.Sys.MemoryMappedProtections.PROT_READ | Interop.Sys.MemoryMappedProtections.PROT_WRITE)) != 0 ? FileAccess.ReadWrite :
-                (protections & (Interop.Sys.MemoryMappedProtections.PROT_WRITE)) != 0 ? FileAccess.Write :
-                FileAccess.Read;
+            FileShare share = inheritability == HandleInheritability.None ?
```

Comment:
is this clearer
```c#
            FileShare share =FileShare.ReadWrite | FileShare.Inheritable;
            if (inheritability == HandleInheritability.None)
                         share ^= FileShare.Inheritable;
```

Reply:
Certainly not the first one :)

The latter seems equally clear to me.  I'll leave it as I have it unless you feel strongly about it.